### PR TITLE
Remove the debug check in JInstaller::abort()

### PR DESCRIPTION
**Pull request by mbabker**
When a site is in debug mode and an extension installation fails, `JInstaller::abort()` will throw an exception (which goes uncaught) and actually complicates debugging errors as now the error stack is related to the exception versus the error which causes the abort method to be called in the first place.  This PR removes the conditional.

### CODE REVIEW

**Comment by mahagr** [comment]
Cannot verify the fix as there's some other change in RC that broke installer on abort.

0 JLIB_INSTALLER_ABORT_INSTALL_CUSTOM_INSTALL_FAILURE 

Is all I get, so it looks like there's exception which wasn't caught..?

**Comment by mbabker** [comment]
Have any extension specifically to test with?  I've mainly tested with my own extensions, Acymailing, and Kunena, and I've not hit that error at all.

**Comment by mbabker** [comment]
Found the issue.  Exceptions being thrown by the method triggering install scripts weren't being caught, causing the uncaught exception to hit the global error handler.  Also traced down the fact that I completely missed adding that language string so it's added in here too.

**Comment by mahagr** [comment]
Sent test files directly in skype. Will test it now.

**Comment by mahagr** [comment]
Nice, confirmed fixing a few different issues:

* Missing translation string
* Debug mode now behaves in the same way as normal mode
* Installation was run even when preflight failed (modified Kunena installer failing on preflight)
* Installation was interrupted without displaying a reason (modified rokgallery failing on preflight)

**Comment by kat05** [comment]
Tested Installation of several of our extensions and templates and can confirm that the patch resolves issues for me too. We REALLY need this to go into J3.4 Stable or else we're in deep trouble!

**Comment by zero-24** [comment]
Thanks for testing! Moving to RTC and add the 3.4.0 milestone (again only on JTracker) :smile:<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6109">issues.joomla.org/joomla-cms/6109</a>.</sub>

**Comment by mbabker** [comment]
@wilsonge This one needs to go in.  Without it, any extension install scripts that return a false results in an uncaught exception bubbling up to the global handler.  Granted, the original PR was for the debug check, but testing this found the exception issue.

**Comment by wilsonge** [comment]
@infograf768 please verify you're happy with these lang strings being added. Personally this isn't functionality that worked before 3.4 so I'm not gonna tear my hair out if we leave this till 3.4.1 (although it's not ideal)

**Comment by mbabker** [comment]
Umm, actually the uncaught exception is a regression from 3.3...

File: administrator/language/en-GB/en-GB.lib_joomla.ini [hunk 1]
```diff
@@ -451,6 +451,7 @@ JLIB_INSTALLER_ABORT_FILE_INSTALL_SQL_ERROR="Files %1$s: SQL error file %2$s"
 JLIB_INSTALLER_ABORT_FILE_ROLLBACK="Files Install: %s"
 JLIB_INSTALLER_ABORT_FILE_SAME_NAME="Files Install: Another extension with same name already exists."
 JLIB_INSTALLER_ABORT_FILE_UPDATE_SQL_ERROR="Files Update: SQL error file %s"
+JLIB_INSTALLER_ABORT_INSTALL_CUSTOM_INSTALL_FAILURE="Extension %s: Custom install routine failure"
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.lib_joomla.ini]: please add a . at the end

File: language/en-GB/en-GB.lib_joomla.ini [hunk 2]
```diff
@@ -451,6 +451,7 @@ JLIB_INSTALLER_ABORT_FILE_INSTALL_SQL_ERROR="Files %1$s: SQL error file %2$s"
 JLIB_INSTALLER_ABORT_FILE_ROLLBACK="Files Install: %s"
 JLIB_INSTALLER_ABORT_FILE_SAME_NAME="Files Install: Another extension with same name already exists."
 JLIB_INSTALLER_ABORT_FILE_UPDATE_SQL_ERROR="Files Update: SQL error file %s"
+JLIB_INSTALLER_ABORT_INSTALL_CUSTOM_INSTALL_FAILURE="Extension %s: Custom install routine failure"
```
Comment (brianteeman) [inline language/en-GB/en-GB.lib_joomla.ini]: please add a . at the end

File: administrator/language/en-GB/en-GB.lib_joomla.ini (hunk 1 above)
Comment (mbabker) [inline administrator/language/en-GB/en-GB.lib_joomla.ini]: Done

File: language/en-GB/en-GB.lib_joomla.ini (hunk 2 above)
Comment (mbabker) [inline language/en-GB/en-GB.lib_joomla.ini]: Done

**Comment by mahagr** [comment]
@infograf768, @wilsonge 

I would rather add those 2 new strings as currently (and in all versions down to J1.6) all users will see this very descriptive message instead:

JLIB_INSTALLER_ABORT_INSTALL_CUSTOM_INSTALL_FAILURE

So the language string has been there from Joomla 1.6... It was just never translated and everyone has seen the above key instead of anything that could be understood.

By not accepting this pull request you will cause major flood of support requests to many developers on subject: "Installation fails on JLIB_INSTALLER_ABORT_INSTALL_CUSTOM_INSTALL_FAILURE".

This is regression from Joomla 3.3 and NEEDS to be fixed.

**Comment by mbabker** [comment]
> JLIB_INSTALLER_ABORT_INSTALL_CUSTOM_INSTALL_FAILURE

Just to note, this string is actually new.  It's based on 6 or 7 other strings (at a minimum) which all have single word differences and my language changes in the installer library have been an attempt at killing off an excess of language strings which only change the extension type or route.

**Comment by mahagr** [comment]
Just tested J!2.5.28 and J!3.3.6 and both do have the translations. I know the translations were missing from some earlier version, but you're right, I do see: 

    Plugin Install: Custom install routine failure

Joomla 3.4.0-RC has no translation, though, and is showing up above key.

**Comment by brianteeman** [comment]
they are different strings.@mbabker is introducing a NEW string with this PR
> JLIB_INSTALLER_ABORT_INSTALL_CUSTOM_INSTALL_FAILURE

The one you just mentioned is
> JLIB_INSTALLER_ABORT_PLG_INSTALL_CUSTOM_INSTALL_FAILURE

NOTE the PLG

**Comment by mahagr** [comment]
You're right, Brian.

The only thing is that new string already exists in RC. I'll PM you in Skype to allow you to test it out.

**Comment by brianteeman** [comment]
So the problem is that the string was added some time ago but it was never added to the language files

**Comment by mbabker** [comment]
For the string, yes.  Considering I added maybe 15-20 generalized strings
to try and replace ~75 needlessly duplicated strings, easy to miss one.

**Comment by brianteeman** [comment]
So as this PR has been accepted and its a bug that the language string is
not in the language file this should be accepted. OF course it would be
better if it was translated but as we have other "error" messages that are
actually hard coded and in english by necessity I dont see a major issue

On 19 February 2015 at 18:06, Michael Babker <notifications@github.com>
wrote:

> For the string, yes. Considering I added maybe 15-20 generalized strings
> to try and replace ~75 needlessly duplicated strings, easy to miss one.
>
> On Thursday, February 19, 2015, Brian Teeman <notifications@github.com>
> wrote:
>
> > So the problem is that the string was added some time ago but it was
> never
> > added to the language files
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/joomla/joomla-cms/pull/6109#issuecomment-75102348>.
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/6109#issuecomment-75103800>.
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by infograf768** [comment]
Folks, I can merge the 2 new strings separately from this PR and let TTs know about it today.
We do take into account this type of last minute new strings.
@mbabker
Shall I ?

**Comment by infograf768** [comment]
Tested with rockfailure. OK for me.
Merging and letting TTs know.

**Comment by wilsonge** [comment]
Yes please JM if that is acceptable

**Pull request merged**